python/lsst/pipe/base/graph/graph.py

```diff
@@ -1262,6 +1262,70 @@ def iterTaskGraph(self) -> Generator[TaskDef, None, None]:
         """
         yield from nx.topological_sort(self.taskGraph)
 
+    def updateRun(self, run: str, *, metadata_key: str | None = None, update_graph_id: bool = False) -> None:
```

Comment:
Higher level code that fills metadata is in [ctrl_mpexec.CmdLienFwk](https://github.com/lsst/ctrl_mpexec/blob/main/python/lsst/ctrl/mpexec/cmdLineFwk.py#L573-L584). I'm more comfortable with having a default here that does not depend on high-level code, and have ctrl_mpexec (which I think will be the only client of this method) to pass the correct value here.

We could potentially define a set of common metadata keys in pipe_base so that they can be usable by other clients, or even make metadata an actual class with a fixed list of common attributes (and optional dynamic mapping).

Reply:
Yeah, if it's CmdLineFwk setting the value in both cases, that's fine, too.